### DISCUSSION TITLE
The inspect method returns the wrong number of Mesh

### DISCUSSION THREAD

**Discussion by neciszhang** [donmccurdy/glTF-Transform#1054] [Q&A] [answered]
Hi Donmccurdy,
[Here is model](https://alsc-nft-shprod-al-oss-1.oss-cn-shanghai.aliyuncs.com/test/%E5%AE%81%E5%A4%8F%E7%BE%8A%E8%82%89.glb);It has 5039 meshes. I use inspect function, it return 26 meshes.

**Comment by donmccurdy** [comment, marked as answer]
Hi @neciszhang! This method returns the correct number of mesh “resources” in the glTF file. I think the difference you're seeing is that several of those meshes are *used within the scene* many times. If you include the 'instances' property on each mesh, you'll get the total you're expecting:

![Screenshot 2023-08-07 at 8 50 45 AM](https://github.com/donmccurdy/glTF-Transform/assets/1848368/56224b96-7076-42f7-9a63-7317f977adb7)

*Screenshot from https://gltf.report/.*

**Reply by neciszhang** [reply]
Hi,
I have calculated that the number of instances is 5035, but the draws in gltf.report is 5039. What is the difference between them?

**Reply by donmccurdy** [reply]
Ultimately any calculation of draw calls is an approximation, the true number depends heavily on the specific renderer's implementation. Instances of meshes are not the same as draw calls necessarily, as a mesh may have several primitives. The implementation provided by glTF Report is equivalent to:

```js
let drawCalls = 0;
for (const mesh of document.getRoot().listMeshes()) {
	const parentCount = mesh.listParents()
		.filter((parent) => parent.propertyType === 'Node')
		.length;
	drawCalls += parentCount * mesh.listPrimitives().length;
}
```

If your model used an extension like `EXT_mesh_gpu_instancing`, then you'd probably factor that into the count as well.
